### ISSUE TITLE
Error when from_email is not provided

### ISSUE THREAD

**Issue by billzhong** [sunscrapers/django-templated-mail#4]
Hi,

I have following codes:

```python
EmailMessage(self.request, {
    'title': 'test',
}, 'email/message.html').send(
    to=['test1@example.com'],
    reply_to=['test2@example.com'],
    bcc=['test3@example.com'],
)
```

But I am getting errors:
```
Traceback:

File "/home/test/env/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
  41.             response = get_response(request)

File "/home/test/env/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  187.                 response = self.process_exception_by_middleware(e, request)

File "/home/test/env/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
  185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

File "/home/test/env/lib/python3.5/site-packages/django/views/decorators/csrf.py" in wrapped_view
  58.         return view_func(*args, **kwargs)

File "/home/test/env/lib/python3.5/site-packages/rest_framework/viewsets.py" in view
  95.             return self.dispatch(request, *args, **kwargs)

File "/home/test/env/lib/python3.5/site-packages/rest_framework/views.py" in dispatch
  494.             response = self.handle_exception(exc)

File "/home/test/env/lib/python3.5/site-packages/rest_framework/views.py" in handle_exception
  454.             self.raise_uncaught_exception(exc)

File "/home/test/env/lib/python3.5/site-packages/rest_framework/views.py" in dispatch
  491.             response = handler(request, *args, **kwargs)

File "/home/test/env/lib/python3.5/site-packages/templated_mail/mail.py" in send
  76.         super(BaseEmailMessage, self).send(*args, **kwargs)

File "/home/test/env/lib/python3.5/site-packages/django/core/mail/message.py" in send
  348.         return self.get_connection(fail_silently).send_messages([self])

File "/home/test/env/lib/python3.5/site-packages/django/core/mail/backends/smtp.py" in send_messages
  111.                 sent = self._send(message)

File "/home/test/env/lib/python3.5/site-packages/django/core/mail/backends/smtp.py" in _send
  123.         from_email = sanitize_address(email_message.from_email, encoding)

File "/home/test/env/lib/python3.5/site-packages/django/core/mail/message.py" in sanitize_address
  161.         address = Address(nm, addr_spec=addr)

File "/usr/lib/python3.5/email/headerregistry.py" in __init__
  42.             a_s, rest = parser.get_addr_spec(addr_spec)

File "/usr/lib/python3.5/email/_header_value_parser.py" in get_addr_spec
  1988.     token, value = get_local_part(value)

File "/usr/lib/python3.5/email/_header_value_parser.py" in get_local_part
  1800.     if value[0] in CFWS_LEADER:

Exception Type: IndexError at /api/test/
Exception Value: string index out of range
```

It was from my production environment, my SMTP service provider is Mailgun.
I couldn't get the same error from my local environment which is console email backend.

Looks like source code [Line 74](https://github.com/sunscrapers/django-templated-mail/blob/master/templated_mail/mail.py#L74) added empty string to `from_email` when it is not provided.
I will make more tests to see if I can catch something.

Thanks.

**Comment by dkainama** [comment]
Hi, I've added this functionality. From source of django.core.mail:
> If from_email is None, use the DEFAULT_FROM_EMAIL setting.

I will add a fix for this issue

https://docs.djangoproject.com/en/2.0/_modules/django/core/mail/

**Comment by pszpetkowski** [comment]
Hi @billzhong 
Thank you for reporting the problem. It should be resolved with the 1.1.1 release, which is already on PyPI.

**Comment by billzhong** [comment]
Thank you!
